### PR TITLE
Paladin: Add Ardent Defender Spellscript

### PR DESCRIPTION
**Pull request by insunaa**
## 🍰 Pullrequest

This PR adds a Spell Script for the Ardent Defender spell.

***It is entirely untested***

### Issues

- fixes https://github.com/cmangos/issues/issues/3207

### How2Test

- Create Paladin
- Skill Ardent Defender in Talent Tree
- Get hit

### Todo / Checklist

- [x] Please test this PR extensively, I couldn't test it for the time being and it might not work at all or do weird things.

### CODE REVIEW

**Comment by insunaa** [comment]
OK, tested the PR and it works as *I* expect. No idea if it fits with the reference however. One thing that remains unknown is the effect that the Character's defense value has on the amount healed.

File: src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp [hunk 1]
```diff
@@ -239,6 +239,33 @@ struct JudgementsOfTheWiseEnergize : public SpellScript
...
+            reduction = aur->GetBasePoints();
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(1)))
+            healMod = aur->GetBasePoints();
+        remainingDamage *= (100 - reduction) / 100.f;
+        if (int32(player->GetHealth()) - remainingDamage > 0 || player->HasAura(66233))
+            return;
+        healMod = player->GetMaxHealth() * (healMod / 100.f);
+        remainingDamage = 0;
+        // ToDo: "(amount healed based on defense)" find out ratios etc.
+        player->CastCustomSpell(player, 66235, &healMod, nullptr, nullptr, TRIGGERED_OLD_TRIGGERED);
```
Comment (killerwife) [inline src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp:264]: Both are self targeted

File: src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp [hunk 2]
```diff
@@ -239,6 +239,33 @@ struct JudgementsOfTheWiseEnergize : public SpellScript
...
+    {
+        currentAbsorb = 0;
+        Player* player = dynamic_cast<Player*>(aura->GetCaster());
+        if (!player)
+            return;
+        if (int32(player->GetHealth()) - remainingDamage > (player->GetMaxHealth() * 0.35))
+            return;
+        int32 reduction = 0;
+        int32 healMod = 0;
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(0)))
```
Comment (killerwife) [inline src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp:254]: Use enum EFFECT_INDEX_0 instead of this

Comment (insunaa) [reply src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp]: done

File: src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp (hunk 1 above)
Comment (insunaa) [reply src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp]: done

File: src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp [hunk 3]
```diff
@@ -239,6 +239,33 @@ struct JudgementsOfTheWiseEnergize : public SpellScript
...
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(EFFECT_INDEX_0)))
+            reduction = aur->GetBasePoints();
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(EFFECT_INDEX_1)))
+            healMod = aur->GetBasePoints();
+        remainingDamage *= (100 - reduction) / 100.f;
+        if (int32(player->GetHealth()) - remainingDamage > 0 || player->HasAura(66233))
+            return;
+        healMod = player->GetMaxHealth() * (healMod / 100.f);
+        remainingDamage = 0;
+        // ToDo: "(amount healed based on defense)" find out ratios etc.
```
Comment (insunaa) [inline src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp:263]: image
this is the only thing I could find and it's not an official resource

File: src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp [hunk 4]
```diff
@@ -239,6 +239,34 @@ struct JudgementsOfTheWiseEnergize : public SpellScript
...
+    {
+        currentAbsorb = 0;
+        Player* player = dynamic_cast<Player*>(aura->GetCaster());
+        if (!player)
+            return;
+        if (int32(player->GetHealth()) - remainingDamage > (player->GetMaxHealth() * 0.35))
+            return;
+        int32 reduction = 0;
+        int32 healMod = 0;
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(EFFECT_INDEX_0)))
```
Comment (killerwife) [inline src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp:254]: Dont need to cast it to SpellEffectIndex anymore :P its redundant

Comment (insunaa) [reply src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp]: done + also added the defense factor (based on flimsy evidence)

**Pull request merged**